mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,485 @@
+            .swap_v2_contracts
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?
+            .maker_swap_v2_contract;
+        let payment_amount = try_tx_s!(wei_from_big_decimal(&args.amount, self.decimals));
+        let payment_args = {
+            let taker_address = public_to_address(args.taker_pub);
+            MakerPaymentArgs {
+                taker_address,
+                taker_secret_hash: try_tx_s!(args.taker_secret_hash.try_into()),
+                maker_secret_hash: try_tx_s!(args.maker_secret_hash.try_into()),
+                payment_time_lock: args.time_lock,
+            }
+        };
+        match &self.coin_type {
+            EthCoinType::Eth => {
+                let data = try_tx_s!(self.prepare_maker_eth_payment_data(&payment_args).await);
+                self.sign_and_send_transaction(
+                    payment_amount,
+                    Action::Call(maker_swap_v2_contract),
+                    data,
+                    U256::from(self.gas_limit_v2.maker.eth_payment),
+                )
+                .compat()
+                .await
+            },
+            EthCoinType::Erc20 {
+                platform: _,
+                token_addr,
+            } => {
+                let data = try_tx_s!(
+                    self.prepare_maker_erc20_payment_data(&payment_args, payment_amount, *token_addr)
+                        .await
+                );
+                self.handle_allowance(maker_swap_v2_contract, payment_amount, args.time_lock)
+                    .await?;
+                self.sign_and_send_transaction(
+                    U256::from(ZERO_VALUE),
+                    Action::Call(maker_swap_v2_contract),
+                    data,
+                    U256::from(self.gas_limit_v2.maker.erc20_payment),
+                )
+                .compat()
+                .await
+            },
+            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                "NFT protocol is not supported for ETH and ERC20 Swaps"
+            ))),
+        }
+    }
+
+    pub(crate) async fn validate_maker_payment_v2_impl(
+        &self,
+        args: ValidateMakerPaymentArgs<'_, Self>,
+    ) -> ValidatePaymentResult<()> {
+        if let EthCoinType::Nft { .. } = self.coin_type {
+            return MmError::err(ValidatePaymentError::ProtocolNotSupported(
+                "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
+            ));
+        }
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .ok_or_else(|| {
+                ValidatePaymentError::InternalError("Expected swap_v2_contracts to be Some, but found None".to_string())
+            })?
+            .maker_swap_v2_contract;
+        validate_payment_args(args.taker_secret_hash, args.maker_secret_hash, &args.amount)
+            .map_to_mm(ValidatePaymentError::InternalError)?;
```

Comment:
we can already validate `taker_secret_hash` and `maker_secret_hash` by enforcing bytes length and using `TryInto` here 
```rust
let taker_secret_hash = args.taker_secret_hash.try_into()?,
let maker_secret_hash = args.maker_secret_hash.try_into()?,`
```

so I believe we have some redundant checks in `validate_payment_args` and I don't think we still need the function. same as other places it's being called too